src/sdl/texture.cpp

```diff
@@ -41,7 +41,9 @@ texture::texture(SDL_Texture* txt)
 	: texture_(txt, &cleanup_texture)
 {
 	if (txt) {
-		SDL_QueryTexture(txt, nullptr, nullptr, &w_, &h_);
+		SDL_PropertiesID props = SDL_GetTextureProperties(txt);
+		w_ = SDL_GetNumberProperty(props, SDL_PROP_TEXTURE_WIDTH_NUMBER, 0);
+		h_ = SDL_GetNumberProperty(props, SDL_PROP_TEXTURE_HEIGHT_NUMBER, 0);
```

Comment:
Looks like we have `SDL_GetTextureSize` now, though that takes floats.... Not sure why? Could either store the w/h as floats (doesn't seem convenient) or do:
```cpp
float w{}, h{};
SDL_GetTextureSize(*this, &w, &h);
std::tie(w_, h_) = std::tuple(static_cast<int>(w), static_cast<int>(h));
```

Reply:
I'm not sure how this is better? It's still three lines of code either way.

Reply:
Yeah, I realized after I wrote that that it's probably cleaner to just consolidate the texture property getters.